### PR TITLE
contract status manager

### PR DESCRIPTION
**Pull request by chaychen2005**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- libprecompiled/UpdateContractStatusPrecompiled.cpp  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4831240&bid=16050948)

File: libprecompiled/Common.h [hunk 1]
```diff
@@ -41,6 +41,9 @@ namespace precompiled
 
 enum PrecompiledError : int
 {
+    // RingSigPrecompiled -51999 ~ -51900
```
Comment (cyjseagull) [inline libprecompiled/Common.h]: Pls modify the comment.

File: libprecompiled/UpdateContractStatusPrecompiled.cpp [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        isValidAddress = false;
+    }
+    return isValidAddress;
+}
+
+bytes UpdateContractStatusPrecompiled::call(
+    ExecutiveContext::Ptr context, bytesConstRef param, Address const& origin)
+{
+    (void)origin;
+    PRECOMPILED_LOG(TRACE) << LOG_BADGE("UpdateContractStatusPrecompiled")
```
Comment (cyjseagull) [inline libprecompiled/UpdateContractStatusPrecompiled.cpp]: use `DEBUG` here.

File: libprecompiled/UpdateContractStatusPrecompiled.cpp [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    {
+        isValidAddress = false;
+    }
+    return isValidAddress;
+}
+
+bytes UpdateContractStatusPrecompiled::call(
+    ExecutiveContext::Ptr context, bytesConstRef param, Address const& origin)
+{
+    (void)origin;
```
Comment (cyjseagull) [inline libprecompiled/UpdateContractStatusPrecompiled.cpp]: Remove this and declare the function as `bytes UpdateContractStatusPrecompiled::call(
    ExecutiveContext::Ptr context, bytesConstRef param, Address const& )` if `origin` is useless here.

File: libprecompiled/Common.h (hunk 1 above)
Comment (chaychen2005) [reply libprecompiled/Common.h]: Done.

File: libprecompiled/UpdateContractStatusPrecompiled.cpp (hunk 2 above)
Comment (chaychen2005) [reply libprecompiled/UpdateContractStatusPrecompiled.cpp]: Done.

File: libprecompiled/UpdateContractStatusPrecompiled.cpp (hunk 3 above)
Comment (chaychen2005) [reply libprecompiled/UpdateContractStatusPrecompiled.cpp]: These params would be used in function later.
I removed some code in order to pass CI.

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4831240&bid=16050948)

File: libprecompiled/UpdateContractStatusPrecompiled.cpp [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    PRECOMPILED_LOG(DEBUG) << LOG_BADGE("UpdateContractStatusPrecompiled")
+                           << LOG_KV("call param", toHex(param));
+
+    // parse function name
+    uint32_t func = getParamFunc(param);
+    bytesConstRef data = getParamData(param);
+
+    dev::eth::ContractABI abi;
+    bytes out;
+    int result = 0;
```
Comment (fisco-dev) [inline libprecompiled/UpdateContractStatusPrecompiled.cpp]: Codacy Issue found: [The scope of the variable 'result' can be reduced.](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4831240)

**Pull request merged**